packages/react-hooks/src/MessagesProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type { RecordsState } from './recordUtils'
+import type { Agent, BaseRecord } from '@aries-framework/core'
+import type { PropsWithChildren } from 'react'
+
+import { useState, createContext, useContext, useEffect } from 'react'
+import * as React from 'react'
+
+import { useBasicMessages, useBasicMessagesByConnectionId } from './BasicMessageProvider'
+import { useCredentialByConnectionId, useCredentials } from './CredentialProvider'
+import { useProofByConnectionId, useProofs } from './ProofProvider'
+
+const MessageContext = createContext<RecordsState<BaseRecord> | undefined>(undefined)
+
+export const useMessages = () => {
+  const messageContext = useContext(MessageContext)
+  if (!MessageContext) {
+    throw new Error('useMessages must be used within a MessageContextProvider')
+  }
+  return messageContext
+}
+
+export const useMessagesByConnectionId = (connectionId: string): BaseRecord[] | undefined => {
```

Comment:
I assumed that messages would mean basic messages that have been exchanged. Should we call this `useExchanges`, or maybe something else that describes it a bit better?

Reply:
I like that idea and think  it is a good clarification.